src/view3d.cpp

```diff
@@ -104,6 +108,24 @@ void View3D::geometryInserted(const QModelIndex& /*unused*/, int firstRow, int l
     geometryChanged();
 }
 
+void View3D::animateViewTransform()
+{
+    m_animatedViewTransformIndex++;
+    double xsi = m_animatedViewTransformIndex/12.0;
+    xsi = xsi*xsi*(3 - 2*xsi);
+
+    m_camera.setCenter(             (1 - xsi)*m_animatedViewTransformStartCamera.center()
+                                       + xsi *m_animatedViewTransformEndCamera.center());
+    m_camera.setRotation(           (1 - xsi)*m_animatedViewTransformStartCamera.rotation()
+                                       + xsi *m_animatedViewTransformEndCamera.rotation());
```

Comment:
This should use proper spherical interpolation with `QQuaternion::slerp()` as a matter of general principle.  (Not really noticeable unless you're doing large rotations, but may as well.)